src/madcast.mli

```diff
@@ -1,6 +1,26 @@
-
 open Parsetree
 
-val derive : core_type -> core_type -> expression list
+exception NoCastFound
+(** Exception raised when no cast can be derived for two types *)
+
+exception SeveralCastFound
+(** Exception raised when more than cast can be derived for two types *)
+
+
+val derive : core_type -> core_type -> expression
 (** Given an input type [itype] and an output type [otype], returns a
-   (possibly empty) list of expressions of type [itype -> otype]. *)
+    casting function of type [itype -> otype].
+    Raise [NoCastFound] if no casting function can be derived
+    Raise [SeveralCastFound] more than one casting function can be derived *)
+
+val split_arrow : core_type -> (core_type * core_type)
+(** [split_arrow ty] returns the domain and co-domain of an arrow type.
+    Raise [Invalid_argument "split_arrow"] if [ty] is not an arrow type *)
+
+val annotate : expression -> core_type -> expression
+(** [annotate expr ty] returns a let expression of the form
+    [let (e : ty) = expr in e] *)
+
+val madcast : core_type -> expression
```

Comment:
I agree to say that that name is really not that good. In particular, it kind of implies that madcast only does that. Which is true for now but might change in the future

Reply:
What about something more informative/verbose like `caster_of_arrow_type` then?

Reply:
I don't know, I was happy with the rather short names we had. Maybe this should be `derive` and `derive` should be something like `find_caster`?

Since this is the all-in-one function, I would like it to have a short "efficient" name. While the others can have longer names

Reply:
Yeah much better!